source/AI.cpp

```diff
@@ -1707,8 +1701,44 @@ void AI::PickUp(Ship &ship, Command &command, const Body &target)
 	
 	// Move toward the target.
 	command.SetTurn(TurnToward(ship, p));
-	if(p.Unit().Dot(ship.Facing().Unit()) > .7)
+	double dp = p.Unit().Dot(ship.Facing().Unit());
+	if(dp > .7)
 		command |= Command::FORWARD;
+	
+	// Use the afterburner if it will not cause you to miss your target.
+	double distance = p.Length();
+	if(command.Has(Command::FORWARD) && ShouldUseAfterburner(ship))
+		if(dp > .9999 || (distance > 1000. && dp > .9)
+				|| (distance > 500. && dp > .975)
+				|| (distance > 250. && dp > .995)
+				|| (distance > 100. && dp > .999))
+			command |= Command::AFTERBURNER;
```

Comment:
Rather than doing this piecewise, I'd probably use a continuous function: `if(dp > max(.9, min(.9999, 1. - distance * distance / 10000000.)))`  would match these numbers almost exactly.

Reply:
Added